torch/lib/THS/generic/THSTensor.c

```diff
@@ -40,6 +54,9 @@ THLongTensor *THSTensor_(indices)(const THSTensor *self) {
 
 THTensor *THSTensor_(values)(const THSTensor *self) {
   if (self->nnz == 0) {
+    if (self->values->nDimension) {
+      THTensor_(resizeNd)(self->values, 0, NULL, NULL);
```

Comment:
Not sure if it's the right way to free memory for empty tensors. Isn't it better to do that directly in `zero_` instead of waiting for someone to call `indices` or `values`?